kafka-client/src/main/java/dev/responsive/kafka/api/stores/ResponsiveStores.java

```diff
@@ -153,6 +153,14 @@ public static <K, V> StoreBuilder<TimestampedKeyValueStore<K, V>> timestampedKey
       final Serde<K> keySerde,
       final Serde<V> valueSerde
   ) {
+    if (storeSupplier instanceof ResponsiveKeyValueBytesStoreSupplier) {
```

Comment:
This is another slightly awkward thing -- ideally we would just have the storeSupplier parameter type be a `ResponsiveKeyValueBytesStoreSupplier` right off the bat. But then we'd have to change the return type of our store supplier APIs to that, and I worry it might confuse people if they return anything other than the plain KeyValueStoreSupplier type that matches what is needed to plug it into Streams.

Obviously if people are following our docs it's fine, but we know not everyone will. Any thoughts or preferences on this choice? cc @agavra

Reply:
c.f. my comment above - I wouldn't block this PR but I think just copying what Stores does is generally the safest bet.